src/main/kotlin/org/rust/ide/hints/RsInlayParameterHintsProvider.kt

```diff
@@ -112,6 +114,29 @@ enum class HintType(desc: String, enabled: Boolean) {
         }
 
         override fun isApplicable(elem: PsiElement): Boolean = elem is RsForExpr
+    },
+
+    /* In case of fields without any expressions
+       PSI is destructed in this place, so only hints before error element will be shown */
+    STRUCT_LITERAL_TYPE_HINT("Show type hints for fields in struct literal", true) {
+        override fun provideHints(elem: PsiElement): List<InlayInfo> {
+            val element = elem as? RsStructLiteral ?: return emptyList()
+            val subst = element.type.typeParameterValues
+            val hints = element.structLiteralBody.structLiteralFieldList.mapNotNull { field ->
+                val text = (field.reference.resolve() as? RsFieldDecl)
+                    ?.typeReference?.type?.substitute(subst)
+                    ?.shortPresentableText
+                    ?: return@mapNotNull null
+                // in case of incomplete field, COLON will be the sibling, not the child
+                val offset =
+                    (field.colon ?: if (field.nextSibling.elementType == RsElementTypes.COLON) field.nextSibling
+                    else field.identifier).textRange.endOffset
+                text to offset
```

Comment:
There was the second comment. You can (I guess) write here directly InlayInfo(text, offset), and then just return `hints` without additional mapping